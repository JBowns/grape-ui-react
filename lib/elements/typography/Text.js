'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Text = undefined;

var _templateObject = _taggedTemplateLiteral(['\n    ', '\n  '], ['\n    ', '\n  ']);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _styledComponents = require('styled-components');

var _styledComponents2 = _interopRequireDefault(_styledComponents);

var _textStyles = require('./textStyles');

var _componentHelpers = require('../../utils/componentHelpers');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _taggedTemplateLiteral(strings, raw) { return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }

var Text = exports.Text = function Text(props) {
  var overrides = Object.assign({}, props, {
    fontSizeBase: '100%'
  });
  var actualBase = (0, _textStyles.textStylesBase)(overrides);
  var ProtoText = _styledComponents2.default.span(_templateObject, actualBase);
  return _react2.default.createElement(ProtoText, (0, _componentHelpers.passThrough)(Text, props));
};

Text.propTypes = {
  color: _propTypes2.default.string,
  fontFamily: _propTypes2.default.string,
  fontWeight: _propTypes2.default.string,
  kerning: _propTypes2.default.string,
  lg: _propTypes2.default.bool,
  sm: _propTypes2.default.bool,
  textAlign: _propTypes2.default.string,
  textDecoration: _propTypes2.default.string
};

Text.defaultProps = {
  color: _textStyles.defaultTextStylesBase.color,
  fontFamily: _textStyles.defaultTextStylesBase.fontFamily,
  fontWeight: _textStyles.defaultTextStylesBase.fontWeight,
  kerning: _textStyles.defaultTextStylesBase.kerning,
  lg: _textStyles.defaultTextStylesBase.lg,
  sm: _textStyles.defaultTextStylesBase.sm,
  textAlign: _textStyles.defaultTextStylesBase.textAlign,
  textDecoration: _textStyles.defaultTextStylesBase.textDecoration
};