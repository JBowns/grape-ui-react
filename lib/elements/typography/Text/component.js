'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.TextComponent = undefined;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _textStyles = require('../textStyles');

var _componentHelpers = require('../../../utils/componentHelpers');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

var TextComponent = function TextComponent(_ref) {
  var children = _ref.children,
      props = _objectWithoutProperties(_ref, ['children']);

  return _react2.default.createElement(
    'span',
    (0, _componentHelpers.passThrough)(TextComponent, props),
    children
  );
};
exports.TextComponent = TextComponent;
TextComponent.propTypes = Object.assign({
  children: _propTypes2.default.any.isRequired
}, _textStyles.typography.propTypes);