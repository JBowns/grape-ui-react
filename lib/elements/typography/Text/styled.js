'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Text = undefined;

var _templateObject = _taggedTemplateLiteral(['\n  ', '\n  ', '\n  ', '\n  ', '\n  ', '\n  ', '\n  ', '\n  ', '\n  ', '\n'], ['\n  ', '\n  ', '\n  ', '\n  ', '\n  ', '\n  ', '\n  ', '\n  ', '\n  ', '\n']);

var _styledComponents = require('styled-components');

var _styledComponents2 = _interopRequireDefault(_styledComponents);

var _styledSystem = require('styled-system');

var _styledHelpers = require('../../../utils/styledHelpers');

var _component = require('./component');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _taggedTemplateLiteral(strings, raw) { return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }

var Text = exports.Text = (0, _styledComponents2.default)(_component.TextComponent)(_templateObject, _styledHelpers.colorCore, _styledHelpers.fontFamilyCore, _styledHelpers.fontSizeCore, _styledSystem.fontWeight, _styledHelpers.letterSpacingCore, _styledHelpers.lineHeightCore, _styledHelpers.fontStyleCore, _styledHelpers.textAlignCore, _styledHelpers.textDecorationCore);

Text.propTypes = Object.assign({}, _styledHelpers.typography.propTypes);

Text.defaultProps = Object.assign({}, _styledHelpers.defaultStylesBase);