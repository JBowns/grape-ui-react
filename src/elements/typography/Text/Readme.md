There are times when inline text needs the a different font property. All of the base styles can be applied to `<Text>`, which can be utilized nested inside any typography element.

```jsx inside Markdown
import { Paragraph } from '../index';

<Paragraph>
  If I want to highlight an individual word
  <Text color="red">red</Text>
  , or make it
  <Text fontWeight="bold">bold</Text>
  , or
  <Text color="red" fontWeight="bold">both</Text>
  , I can just use the inline <code>&lt;Text&gt;</code> element.
</Paragraph>
```
