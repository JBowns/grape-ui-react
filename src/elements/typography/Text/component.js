import React from 'react';
import PropTypes from 'prop-types';
import { spaceProps, typography } from 'src/utils/styledHelpers';
import { removeSomeProps } from 'src/utils/componentHelpers';

const propsToTrim = {
  ...spaceProps.propTypes,
  ...typography.propTypes,
};

export const TextComponent = ({ children, ...props }) => (
  <span {...removeSomeProps(props, Object.keys(propsToTrim))}>
    {children}
  </span>
);

TextComponent.propTypes = {
  children: PropTypes.any.isRequired,
};
