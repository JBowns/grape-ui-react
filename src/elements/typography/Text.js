import styled from 'styled-components';
import { textStylesBase } from './textStyles';

export const Text = styled.span`
  ${textStylesBase}
`;
