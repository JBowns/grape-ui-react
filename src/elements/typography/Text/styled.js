import styled from 'styled-components';
import { fontWeight, space } from 'styled-system';
import {
  colorCore,
  defaultStylesBase,
  ellipsisCore,
  fontFamilyCore,
  fontSizeCore,
  fontStyleCore,
  letterSpacingCore,
  lineHeightCore,
  textAlignCore,
  textDecorationCore,
  typography,
} from 'src/utils/styledHelpers';
import { TextComponent } from './component';

export const Text = styled(TextComponent)`
  ${colorCore}
  ${ellipsisCore}
  ${fontFamilyCore}
  ${fontSizeCore}
  ${fontStyleCore}
  ${fontWeight}
  ${letterSpacingCore}
  ${lineHeightCore}
  ${textAlignCore}
  ${textDecorationCore}
  ${space}
`;

Text.propTypes = {
  ...typography.propTypes,
};

Text.defaultProps = {
  ...defaultStylesBase,
};
