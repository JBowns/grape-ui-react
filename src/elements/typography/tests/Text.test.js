import React from 'react';
import { Text } from '../Text';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

describe('Text Component', () => {
  it('should return a Text object with base styling', () => {
    const component = renderer.create(<Text>Lorem Ipsum</Text>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should return a Text object with font-style: italic', () => {
    const component = renderer.create(<Text italic>Lorem Ipsum</Text>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should return a Text object with font-size: lg', () => {
    const component = renderer.create(<Text lg>Lorem Ipsum</Text>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should return a Text object with font-size: sm', () => {
    const component = renderer.create(<Text sm>Lorem Ipsum</Text>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should return a Text object with font-family: serif', () => {
    const component = renderer.create(<Text fontFamily="serif">Lorem Ipsum</Text>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should return a Text object with kerning: 1px', () => {
    const component = renderer.create(<Text kerning="1px">Lorem Ipsum</Text>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should return a Text object with text-align: right', () => {
    const component = renderer.create(<Text textAlign="right">Lorem Ipsum</Text>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
