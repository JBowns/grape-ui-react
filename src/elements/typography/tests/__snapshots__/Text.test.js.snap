// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text Component should return a Text object with base styling 1`] = `
.c0 {
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: calc(1rem * 1);
  font-weight: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  color: inherit;
  text-align: inherit;
}

<span
  className="c0"
>
  Lorem Ipsum
</span>
`;

exports[`Text Component should return a Text object with font-family: serif 1`] = `
.c0 {
  font-family: Frutiger,"Frutiger Linotype",Univers,Calibri,"Gill Sans","Gill Sans MT","Myriad Pro",Myriad,"DejaVu Sans Condensed","Liberation Sans","Nimbus Sans L",Tahoma,Geneva,"Helvetica Neue",Helvetica,Arial,sans-serif;
  font-size: calc(1rem * 1);
  font-weight: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  color: inherit;
  text-align: inherit;
}

<span
  className="c0"
  fontFamily="serif"
>
  Lorem Ipsum
</span>
`;

exports[`Text Component should return a Text object with font-size: lg 1`] = `
.c0 {
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: calc(1rem * 1.2);
  font-weight: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  color: inherit;
  text-align: inherit;
}

<span
  className="c0"
>
  Lorem Ipsum
</span>
`;

exports[`Text Component should return a Text object with font-size: sm 1`] = `
.c0 {
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: calc(1rem * .8);
  font-weight: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  color: inherit;
  text-align: inherit;
}

<span
  className="c0"
>
  Lorem Ipsum
</span>
`;

exports[`Text Component should return a Text object with font-style: italic 1`] = `
.c0 {
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: calc(1rem * 1);
  font-weight: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  font-style: italic;
  color: inherit;
  text-align: inherit;
}

<span
  className="c0"
>
  Lorem Ipsum
</span>
`;

exports[`Text Component should return a Text object with kerning: 1px 1`] = `
.c0 {
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: calc(1rem * 1);
  font-weight: inherit;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  line-height: 1.5;
  color: inherit;
  text-align: inherit;
}

<span
  className="c0"
  kerning="1px"
>
  Lorem Ipsum
</span>
`;
